src/ocean/core/Option.d

```diff
@@ -0,0 +1,236 @@
+/******************************************************************************
+
+    Wraps any type in a struct that also contains boolean field indicating if
+    value is in defined state.
+
+    If T is a value type, `Option!(T)` is value type too.
+
+    Copyright:
+        Copyright (c) 2009-2016 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+
+*******************************************************************************/
+
+module ocean.core.Option;
+
+version(UnitTest)
+{
+    import ocean.core.Test;
+}
+
+/// ditto
+struct Option ( T )
```

Comment:
Why `Option` not `Optional`? The word "option" makes me think of a choice between an arbitrary number of things (A, B, C, etc...), whereas "optional" implies a single thing that may or may not be present.

Reply:
I am in doubts. `Option` is how this concept is usually named in various functional languages (say, Haskell). But from language PoV indeed `Optional` makes more sense.

Reply:
@nemanja-boric-sociomantic and @mathias-lang-sociomantic :+1:ed my original comment. Does that mean they also prefer `Optional`?

Reply:
Probably, but I want explicit confirmation that we prefer common sense over widespread naming :)

Reply:
Right, I'm asking them for confirmation.

Reply:
Yeah, for me `Option` also implies "take one from the set". IIRC, Haskel calls it `Maybe`, Rust calls it `Option`, and C++ calls it `std::optional`. Common sense tells me that the C++ got it right :-)

Reply:
I say screw it all and let's go with `Maybe` :smile: